test/gtest/avgpool.hpp

```diff
@@ -0,0 +1,449 @@
+        }
+        else if(dims == 5)
+        {
+            cpu_avgpool_forward_3d<T>(input,
+                                      ref_output,
+                                      static_cast<long>(N),
+                                      static_cast<long>(C),
+                                      static_cast<long>(D),
+                                      static_cast<long>(H),
+                                      static_cast<long>(W),
+                                      static_cast<long>(OD),
+                                      static_cast<long>(OH),
+                                      static_cast<long>(OW),
+                                      ksize_long,
+                                      stride_long,
+                                      padding_long,
+                                      count_include_pad,
+                                      divisor_override);
+        }
+        status = miopen::avgpool::AvgPoolForward(handle,
+                                                 input.desc,
+                                                 input_dev.get(),
+                                                 output.desc,
+                                                 output_dev.get(),
+                                                 ksize.GetSize() == 3 ? ksize[0] : 1,
+                                                 ksize.GetSize() == 3 ? ksize[1] : ksize[0],
+                                                 ksize.GetSize() == 3 ? ksize[2] : ksize[1],
+                                                 stride.GetSize() == 3 ? stride[0] : 1,
+                                                 stride.GetSize() == 3 ? stride[1] : stride[0],
+                                                 stride.GetSize() == 3 ? stride[2] : stride[1],
+                                                 padding.GetSize() == 3 ? padding[0] : 1,
+                                                 padding.GetSize() == 3 ? padding[1] : padding[0],
+                                                 padding.GetSize() == 3 ? padding[2] : padding[1],
+                                                 count_include_pad,
+                                                 divisor_override);
+        fflush(stdout);
+        ASSERT_EQ(status, miopenStatusSuccess);
+
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<T>::epsilon();
+
+        auto error = miopen::rms_range(ref_output, output);
+
+        ASSERT_EQ(miopen::range_distance(ref_output), miopen::range_distance(output));
+        EXPECT_LT(error, threshold * 10);
+    }
+    AvgPoolTestCase avgpool_config;
+
+    tensor<T> input;
+    tensor<T> output;
+    tensor<T> ref_output;
+    tensor<int64_t> ksize;
+    tensor<long> ksize_long;
+    tensor<int64_t> stride;
+    tensor<long> stride_long;
+    tensor<int64_t> padding;
+    tensor<long> padding_long;
```

Comment:
May I ask why do you use different types (int64_t vs long) for these fields?

Reply:
@hieule88 I can use `tensor<uint64_t>` inside cpu gtest file. Here is the link https://github.com/ROCm/MIOpen/pull/3166/files#diff-fea5a482ae997433e9e5c37e24d18bac565a3f6d0b0d64b6de8fe3027973c353

Reply:
resolved